app/screens/create_or_edit_channel/index.tsx

```diff
@@ -0,0 +1,280 @@
+    };
+
+    const close = (goBack = false): void => {
+        Keyboard.dismiss();
+        if (!isDirect()) {
+        //     this.props.actions.setChannelDisplayName(this.state.displayName);
+        }
+
+        if (goBack) {
+            popTopScreen();
+        } else {
+            dismissModal();
+        }
+    };
+
+    const [appState, dispatch] = useReducer((state: RequestState, action: RequestAction) => {
+        switch (action.type) {
+            case RequestActions.START:
+                emitCanSaveChannel(true);
+                Keyboard.dismiss();
+                return {
+                    ...state,
+                    error: '',
+                    saving: true,
+                };
+            case RequestActions.COMPLETE:
+                emitCanSaveChannel(false);
+                close(true);
+
+                // TODO: setChannelDisplayName
+                return {
+                    ...state,
+                    error: '',
+                    saving: false,
+                };
+            case RequestActions.FAILURE:
+                emitCanSaveChannel(false);
+                return {
+                    ...state,
+                    error: action.error,
+                    saving: false,
+                };
+
+            default:
+                return {
+                    ...state,
+                };
+        }
+    }, {
+        error: '',
+        saving: false,
+        type: channel?.type as ChannelType || General.OPEN_CHANNEL,
+    });
+
+    useEffect(() => {
+        const button = {
+            testID: 'edit_channel.save.button',
+            id: EDIT_CHANNEL_ID,
+            enabled: false,
+            showAsAction: 'always',
+            color: theme.sidebarHeaderTextColor,
+            text: formatMessage({id: 'mobile.edit_channel', defaultMessage: 'Save'}),
+        };
+
+        if (!channel) {
+            button.id = CREATE_CHANNEL_ID;
+            button.text = formatMessage({id: 'mobile.create_channel', defaultMessage: 'Create'});
+        }
+
+        setRightButton(button);
+        setButtons(componentId, {
+            leftButtons: [],
+            rightButtons: [{...button, enabled: false}] as never[],
+        });
+    }, []);
+
+    useEffect(() => {
+        const update = Navigation.events().registerComponentListener({
+            navigationButtonPressed: ({buttonId}: {buttonId: string}) => {
+                switch (buttonId) {
+                    case CLOSE_CHANNEL_ID:
+                        close(true);
+                        break;
+                    case CREATE_CHANNEL_ID:
+                        onCreateChannel();
+                        break;
+                    case EDIT_CHANNEL_ID:
+                        onUpdateChannel();
+                        break;
+                }
+            },
+        }, componentId);
+
+        return () => {
+            update.remove();
+        };
+    }, [displayName, header, purpose]);
```

Comment:
0/5 I feel is hard to read and maintain the dependencies like this. I guess that displayName, header and purpose are values used in the onCreateChannel or onUpdateChannel functions. I like it better when we just depend on the functions, and those functions are the ones depending on the actual values. But it is just a personal preference which may not be useful here.

Reply:
this looks fine to me

Reply:
Looking closer at the latest commit, this effect now depends on the two `useCallback`s `onCreateChannel` and `onUpdateChannel` instead of the deps for those two callbacks. This is cleaner because the effect is dependant on the callbacks and it is not necessary to know what the dependancies of those callbacks are..  Just use the callbacks as a dependant